src/big/big_decimal.cr

```diff
@@ -0,0 +1,253 @@
+require "big_int"
+
+# A `BigDecimal` represents arbitrary precision decimals.
+#
+# It is internally represented by a pair of `BigInt` and `UInt64`: value and scale.
+# Value contains the actual value, and scale tells the decimal point place.
+# e.g. value=1234, scale=2 => 12.34
+#
+# The general idea and some of the arithmetic algorithms were adapted from
+# the MIT/APACHE -licensed https://github.com/akubera/bigdecimal-rs
+
+class InvalidBigDecimalException < Exception
+end
+
+struct BigDecimal
+  ZERO                       = BigInt.new(0)
+  TEN                        = BigInt.new(10)
+  DEFAULT_MAX_DIV_ITERATIONS = 100_u64
```

Comment:
Personally I wouldn't bother making this a constant but it's not worth any effort making it not a constant.

So ignore this comment.